### PR TITLE
Internationalization

### PR DESCRIPTION
**Pull request by RobinJadoul**
Started because of #58 
I hope I didn't miss any :-)

Still to do: add instructions on helping out with translation to the README

### CODE REVIEW

**Comment by EMLVIRUS** [comment]
I suggest the guidelines should include text used on AMO and chrome store pages.

**Comment by LEDfan** [comment]
Good job! I rebased and added the text to the README as discussed.

 @RobinJadoul I also added the Dutch translations, but please give them a good review :smiley:

File: README.md [hunk 1]
```diff
@@ -62,6 +62,30 @@ When Keepass finds multiple Credentials for one URL, it will ask you to select o
...
+ - NL
+ 
+To provide extra translations please open a pull request:
+ 1. fork this repo and clone it into your workspace
+ 2. copy the `_locales/en/messages.json` file to e.g. `_locales/de/message.json`
+ 3. translate all `message` fields inside this file
+ 4. commit and push to Github
+ 5. open a pull request
+ 
+We believe that the following words should not be translated, because they are well-known and improves the UX:
```
Comment (LEDfan) [inline README.md]: Agree @RobinJadoul ?

Comment (RobinJadoul) [reply README.md]: I'd rather just restrict this to Keywi and Keepass
database and hash may have proper translations, while Secure Storage isn't really well known nor would we want it to given we'd like to replace it with FF's password manager

File: _locales/nl/messages.json [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  "extensionDescription": {
+    "message": "Minimalistische Keepass plugin gebaseerd op Web Extensions en KeepassHTTP",
+    "description": "The description for the extension, used in the manifest."
+  },
+  "fillUsernamePassDescription": {
+    "message": "Vul uw gebruikersnaam en paswoord vanuit het huidige actieve invoerveld",
+    "description": "The description for the fill username/password hotkey"
+  },
+  "fillPassDescription": {
+    "message": "Vul uw paswoord in het hduige active invoerveld",
```
Comment (RobinJadoul) [inline _locales/nl/messages.json]: typo

maybe just keep 'huidige' or 'actieve', but not both

File: _locales/nl/messages.json [hunk 3]
```diff
@@ -0,0 +1,182 @@
+{
+  "extensionDescription": {
+    "message": "Minimalistische Keepass plugin gebaseerd op Web Extensions en KeepassHTTP",
+    "description": "The description for the extension, used in the manifest."
+  },
+  "fillUsernamePassDescription": {
+    "message": "Vul uw gebruikersnaam en paswoord vanuit het huidige actieve invoerveld",
```
Comment (RobinJadoul) [inline _locales/nl/messages.json]: - see below for 'huidige actieve'
- I think '... invullen' would be better than 'Vul ... in'

File: _locales/nl/messages.json [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+  "SSunlockCancelled": {
+    "message": "Secure Storage ontgrendelen werd geannuleerd, Keywi kan niet functioneren zonder deze stap.",
+    "description": "Displayed when the unlock for the Secure Storage is cancelled."
+  },
+  "SSwrongUnlockKey": {
+    "message": "U heeft een foute key ingevoerd.",
+    "description": "Alert when the wrong key is given at Secure Storage unlock time"
+  },
+  "contextFillUserPass": {
+    "message": "Vul gebruikersnaam and paswoord in",
```
Comment (RobinJadoul) [inline _locales/nl/messages.json]: invullen

File: _locales/nl/messages.json [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+    "description": "Context menu to fill username and password"
+  },
+  "contextFillUser": {
+    "message": "Vul gebruikersnaam in",
```
Comment (RobinJadoul) [inline _locales/nl/messages.json]: invullen

File: _locales/nl/messages.json [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+    "description": "Context menu to fill username"
+  },
+  "contextFillPass": {
+    "message": "Vul paswoord in",
```
Comment (RobinJadoul) [inline _locales/nl/messages.json]: Invullen

File: _locales/nl/messages.json [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+    "description": "Context menu to fill password"
+  },
+  "noPassFieldFound": {
+    "message": "Geen paswoord invoerveld gevonden, enkel de gebruikersnaam word ingevuld",
```
Comment (RobinJadoul) [inline _locales/nl/messages.json]: - dt-fout
- maybe just 'paswoordveld'

File: _locales/nl/messages.json [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+  "assocFailed": {
+    "message": "Er ging iets mis tijdens de associatie met Keepass",
+    "description": "Displayed when something goes wrong during association"
+  },
+  "statusSSunlocked": {
+    "message": "Secure storage ontgrendeld",
+    "description": "is Secure Storage unlocked?"
+  },
+  "statusDBassoc": {
+    "message": "Keepass database geassocieerd",
```
Comment (RobinJadoul) [inline _locales/nl/messages.json]: 'geassocieerd met ...' ?

File: _locales/nl/messages.json [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+  "statusDBhash": {
+    "message": "Keepass database hash",
+    "description": "What is the DB hash?"
+  },
+  "statusDBid": {
+    "message": "Keepass database id",
+    "description": "What is the DB id?"
+  },
+  "confirmNotPassField": {
+    "message": "Dit is geen paswoord veld, bent u zeker om hier uw paswoord in te vullen?",
```
Comment (RobinJadoul) [inline _locales/nl/messages.json]: 'wilt u hier echt'?

File: _locales/nl/messages.json [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+    "description": "Confirm with the user they want to fill a password in something that is no password field"
+  },
+  "dialogSetupSSTitle": {
+    "message": "Stel de Secure Storage in",
```
Comment (RobinJadoul) [inline _locales/nl/messages.json]: 'Secure Storage instellen'?

File: _locales/nl/messages.json [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+    "description": "The title for the setup Secure Storage dialog"
+  },
+  "dialogSetupSSHeader": {
+    "message": "Stel de Secure Storage in",
```
Comment (RobinJadoul) [inline _locales/nl/messages.json]: 'Secure Storage instellen'?

File: _locales/nl/messages.json [hunk 12]
```diff
@@ -0,0 +1,182 @@
...
+  "dialogSelectHeader": {
+    "message": "Selecteer een paswoord",
+    "description": "The header of the select multiple passwords dialog"
+  },
+  "dialogSelectBody": {
+    "message": "Keepass heeft meerder paswoorden gevonden voor deze URL, gelieve er één te selecteren.",
+    "description": "The body of the select multiple passwords dialog"
+  },
+  "Cancel": {
+    "message": "Annuleer",
```
Comment (RobinJadoul) [inline _locales/nl/messages.json]: 'Annuleren'?
cf. 'opslaan'

File: _locales/nl/messages.json [hunk 13]
```diff
@@ -0,0 +1,182 @@
...
+  "Save": {
+    "message": "Opslaan",
+    "description": "Save"
+  },
+  "Password": {
+    "message": "Paswoord",
+    "description": "Password"
+  },
+  "samePass": {
+    "message": "Herhaal paswoord",
```
Comment (RobinJadoul) [inline _locales/nl/messages.json]: 'het paswoord'?

File: _locales/nl/messages.json [hunk 14]
```diff
@@ -0,0 +1,182 @@
...
+    "description": "Same password"
+  },
+  "Unlock": {
+    "message": "Ontgrendel",
```
Comment (RobinJadoul) [inline _locales/nl/messages.json]: 'Ontgrendelen'?

File: _locales/nl/messages.json [hunk 15]
```diff
@@ -0,0 +1,182 @@
...
+  "optKeepassHTTPURL": {
+    "message": "KeepassHTTP url:",
+    "description": "options page: KeepassHTTP url"
+  },
+  "optHashRounds": {
+    "message": "Aantal hashing rounds voor key generating (geavanceerd)",
+    "description": "options page: number of hashing rounds"
+  },
+  "optDeferUnlockSS": {
+    "message": "Ontgrendel de Secure Storage niet bij het opstarten van Firefox",
```
Comment (RobinJadoul) [inline _locales/nl/messages.json]: 'Secure Storage niet ontgrendelen ...' ?

**Comment by RobinJadoul** [comment]
And for the metadata/readme: localized screenshots or is that too much trouble (having users build and testdrive the plugin themselves)?

File: _locales/nl/messages.json (hunk 8 above)
Comment (LEDfan) [reply _locales/nl/messages.json]: selectie_704
This is text is shown on the options page, I think it's okay like this?

**Comment by LEDfan** [comment]
> And for the metadata/readme: localized screenshots or is that too much trouble (having translators build and testdrive the plugin themselves)?

I think that's too much trouble.

**Comment by EMLVIRUS** [comment]
Sorry to interrupt here. @LEDfan Do you indicate that you will keep stings on AMO English rather that localized?

**Comment by LEDfan** [comment]
> Sorry to interrupt here. @LEDfan Do you indicate that you will keep strings on AMO English rather than being localized?

No, the string on AMO can be translated, but we will not provide localized screenshots.

File: README.md [hunk 16]
```diff
@@ -83,6 +91,20 @@ The following word should always be capitalized:
...
+
+### Translation of addons.mozilla.org metadata
+ 1. fork this repo and clone it into your workspace
+ 2. copy the `_amo/en/description.html` file to e.g. `_amo/nl/description.html`
+ 2. copy the `_amo/en/summary.html` file to e.g. `_amo/nl/summary.html`
+ 3. translate it
+ 4. commit and push to Github
+ 5. open a pull request
+
+In `description.html` these HTML tags are supported:
```
Comment (RobinJadoul) [inline README.md]: What about `summary.html`, maybe explicit list that one as well, or just refer to 'the html files'?

Comment (LEDfan) [reply README.md]: The summary actually only supports text, I updated the README

**Pull request merged**